frontend/src/components/Semester/SemesterList/SemesterCard.js

```diff
@@ -0,0 +1,157 @@
+        showConfirmDialog,
+        setIsOpenSemesterCopyForm,
+        setSemesterId,
+        createArchivedSemester,
+        setSemesterGroupsOptions,
+        setIsOpenGroupsDialog,
+        semDays,
+        semester,
+    } = props;
+    const { t } = useTranslation('formElements');
+
+    return (
+        <Card
+            additionClassName={`semester-card done-card ${
+                semester.currentSemester ? 'current' : ''
+            }`}
+        >
+            <div className="cards-btns">
+                {!(disabled || archived) && (
+                    <>
+                        <IoMdEye
+                            className="svg-btn copy-btn"
+                            title={t(COMMON_SET_DISABLED)}
+                            onClick={() => {
+                                showConfirmDialog(semester.id, dialogTypes.SET_VISIBILITY_DISABLED);
+                            }}
+                        />
+                        <FaEdit
+                            className="svg-btn edit-btn"
+                            title={t(EDIT_TITLE)}
+                            onClick={() => {
+                                selectSemesterSuccess(semester.id);
+                            }}
+                        />
+                        <FaCopy
+                            className="svg-btn copy-btn"
+                            title={t(COPY_LABEL)}
+                            onClick={() => {
+                                setIsOpenSemesterCopyForm(true);
+                                setSemesterId(semester.id);
+                            }}
+                        />
+                        {!semester.currentSemester && (
+                            <FaFileArchive
+                                className="svg-btn archive-btn"
+                                title={t(COMMON_MAKE_ARCHIVE)}
+                                onClick={() => {
+                                    createArchivedSemester(semester.id);
+                                }}
+                            />
+                        )}
+                    </>
+                )}
+                {disabled && !archived && (
+                    <GiSightDisabled
+                        className="svg-btn copy-btn"
+                        title={t(COMMON_SET_ENABLED)}
+                        onClick={() => {
+                            showConfirmDialog(semester.id, dialogTypes.SET_VISIBILITY_ENABLED);
+                        }}
+                    />
+                )}
+                {/* {archived && (
+                                <IoMdEye
+                                    className="svg-btn copy-btn"
+                                    title={t(COMMON_PREVIEW)}
+                                    onClick={() => {
+                                        handleSemesterArchivedPreview(semester.id);
+                                    }}
+                                />
+                            )} */}
```

Comment:
If you don't need this code, delete it

Reply:
it will need in future for implement archived semester functionality